src/libraries/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -38,9 +38,25 @@ public sealed partial class Lock
         private uint _state; // see State for layout
         private uint _recursionCount;
         private short _spinCount;
-        private ushort _waiterStartTimeMs;
+
+        // The lowest bit is a flag, when set it indicates that the lock should not use alertable waits
+        private ushort _waiterStartTimeMsAndFlags;
+
         private AutoResetEvent? _waitEvent;
 
+#if NATIVEAOT // The method needs to be public in NativeAOT so that other private libraries can access it
+        public Lock(bool useAlertableWaits)
+#else
+        internal Lock(bool useAlertableWaits)
```

Comment:
```suggestion
        public Lock(bool forceTrivialWaitOperations)
#else
        internal Lock(bool forceTrivialWaitOperations)
```

Reply:
That name sounds good for a global option. For an object instance, maybe it could be simpler, how about `useTrivialWaits` or `forceTrivialWaits`?

Reply:
Also not sure if "trivial waits" would imply not spin-waiting, that may be something else that we'd want to expose in the future, so maybe it could be more clear.

Reply:
If we were to expose something for spin-waiting I guess it would be a spin count, so maybe there wouldn't be much confusion. I'll start with `forceTrivialWaits` for now.

Reply:
> useTrivialWaits or forceTrivialWaits?

Sounds good to me. (I like `useTrivialWaits` a bit more out of these two choices.)